samples/glean/src/main/java/org/mozilla/samples/glean/GleanApplication.kt

```diff
@@ -37,10 +38,13 @@ class GleanApplication : Application() {
         val config = Configuration(httpClient = httpClient)
         Glean.initialize(applicationContext, uploadEnabled = true, configuration = config)
 
-        // Initialize the Experiments library and pass in the callback that will generate a
+        // Initialize the Nimbus experiments library and pass in the callback that will generate a
         // broadcast Intent to signal the application that experiments have been updated. This is
-        // only relevant to the experiments library, aside from recording the experiment in Glean.
-        Experiments.initialize(applicationContext, ExperimentsConfig(httpClient = client)) {
+        // only relevant to the Nimbus library, aside from recording the experiment in Glean.
+        RustLog.enable()
+        RustHttpConfig.setClient(lazy { HttpURLConnectionClient() })
```

Comment:
This seems strange to me: having the application give both the `RustHttpConfig` and Glean a `lazy { HttpURLConnectionClient() }`.

Is the `RustHttpConfig` client not used in Glean? Should that sort of config be done here, rather than in `Nimbus.initialize`? I think I'm missing something in my understanding.

Reply:
I believe the networking in Glean happens in the Kotlin code rather than the Rust code.

From an "idiomatic a-c code" perspective, `RustHttpConfig` is the odd one out here in the way that you have to configure a global Client, while all pure Kotlin consumers are expected to take a `Client` in their constructor or similar.

Reply:
Ah, ok. Gotcha.